test/resqml2_0_1test/AbstractBigIjkGridRepresentationTest.cpp

```diff
@@ -0,0 +1,223 @@
+
+				nodesIjkGridRepresentation[3 * currentIndex] = i * xIncr;
+				nodesIjkGridRepresentation[3 * currentIndex + 1] = j * yIncr;
+				nodesIjkGridRepresentation[3 * currentIndex + 2] = k * zIncr + i * faultThrow;
+			}
+
+	return nodesIjkGridRepresentation;
+}
+
+void AbstractBigIjkGridRepresentationTest::initSplitCoordinateLine(unsigned int * pillarOfCoordinateLine,
+	unsigned int * splitCoordinateLineColumnCumulativeCount,
+	unsigned int * splitCoordinateLineColumns)
+{
+	//*************************************
+	// initializing  pillarOfCoordinateLine
+	
+	unsigned int currentIndex = 0;
+	for (unsigned int j = 0; j < jCount + 1; ++j)
+		for (unsigned int i = 1; i < faultCount + 1; ++i)
+		{
+			pillarOfCoordinateLine[currentIndex] = j * (iCount + 1) + i;
+			++currentIndex;
+		}
+
+	//******************************************************
+	// initializing splitCoordinateLineColumnCumulativeCount
+
+	unsigned int currentCumulativeCount = 0;
+	for (unsigned int faultIndex = 0; faultIndex < faultCount; ++faultIndex)
+	{
+		++currentCumulativeCount;
+		splitCoordinateLineColumnCumulativeCount[faultIndex] = currentCumulativeCount;
+	}
+
+	for (unsigned int jIndex = 1; jIndex < jCount; ++jIndex)
+	{
+		for (unsigned int faultIndex = 0; faultIndex < faultCount; ++faultIndex)
+		{
+			currentCumulativeCount += 2;
+			splitCoordinateLineColumnCumulativeCount[jIndex * faultCount + faultIndex] = currentCumulativeCount;
+		}
+	}
+
+	for (unsigned int faultIndex = 0; faultIndex < faultCount; ++faultIndex)
+	{
+		++currentCumulativeCount;
+		splitCoordinateLineColumnCumulativeCount[jCount * faultCount + faultIndex] = currentCumulativeCount;
+	}
+
+	//****************************************
+	// initializing splitCoordinateLineColumns
+
+	currentIndex = 0;
+	for (unsigned int faultIndex = 0; faultIndex < faultCount; ++faultIndex)
+	{
+		splitCoordinateLineColumns[currentIndex] = faultIndex + 1;
+		++currentIndex;
+	}
+
+	for (unsigned int jIndex = 1; jIndex < jCount; ++jIndex)
+		for (unsigned int faultIndex = 0; faultIndex < faultCount; ++faultIndex)
+		{
+			splitCoordinateLineColumns[currentIndex] = ((jIndex - 1) * iCount) + faultIndex + 1;
+			splitCoordinateLineColumns[currentIndex + 1] = (jIndex * iCount) + faultIndex + 1;
+			currentIndex += 2;
+		}
+
+	for (unsigned int faultIndex = 0; faultIndex < faultCount; ++faultIndex)
+	{
+		splitCoordinateLineColumns[currentIndex] = ((jCount - 1) * iCount) + faultIndex + 1;
+		++currentIndex;
+	}
+}
+
+void AbstractBigIjkGridRepresentationTest::initDiscreteProperty(unsigned short * discretePropertyValues)
+{
+	for (unsigned short k = 0; k < kCount; ++k)
+		for (unsigned int j = 0; j < jCount; ++j)
+			for (unsigned int i = 0; i < iCount; ++i)
+			{
+				discretePropertyValues[k * iCount * jCount + j * iCount + i] = k;
+			}
+}
+
+void AbstractBigIjkGridRepresentationTest::initContinuousProperty(double * continuousPropertyValues)
+{
+	double valueIncr;
```

Comment:
I think it would be more readable using ? : construct.

Reply:
I do not understand this one. Can you please rephrase?

Reply:
I think it is named "conditional operator" : https://www.tutorialspoint.com/cplusplus/cpp_conditional_operator.htm
It shorten a lot a simple "if then else" statement.

Reply:
Ok!